hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/container/ozoneimpl/TestOzoneContainer.java

```diff
@@ -307,9 +291,18 @@ static void runTestBothGetandPutSmallFile(
           = ContainerTestHelper.getReadSmallFileRequest(client.getPipeline(),
           smallFileRequest.getPutSmallFile().getBlock());
       response = client.sendCommand(getSmallFileRequest);
-      Assert.assertArrayEquals(
-          smallFileRequest.getPutSmallFile().getData().toByteArray(),
-          response.getGetSmallFile().getData().getData().toByteArray());
+
+      Assert.assertEquals(ContainerProtos.Result.SUCCESS, response.getResult());
+
+      ContainerProtos.ReadChunkResponseProto chunkResponse =
+          response.getGetSmallFile().getData();
+      if (chunkResponse.hasDataBuffers()) {
+        Assert.assertArrayEquals(requestBytes,
+            chunkResponse.getDataBuffers().toByteArray());
+      } else {
+        Assert.assertArrayEquals(requestBytes,
+            chunkResponse.getData().toByteArray());
+      }
```

Comment:
This method is called only from``testBothGetandPutSmallFile`` which we have disabled now and I suppose it is to be chased as part of HDDS-6473, so why are we making changes to this method here?

The test fails also in this method only...
```
array lengths differed, expected.length=1024 actual.length=0; arrays first differed at element [0]; expected:<27> but was:<end of array>
Expected :27
Actual   :end of array
```

Reply:
ReadChunk response has two versions: one with single data array, one with multiple buffers.  Initially I thought that the test is failing because it checks `data` but response has `dataBuffer`.  This turned out to be false, but I still think handling both versions makes the test less fragile.

In its current form the test reproduces HDDS-6473 with the failure you mentioned.

Reply:
Makes sense.
I see [HDDS-6473](https://issues.apache.org/jira/browse/HDDS-6473) is on your name only. Do get that in and resolve this. :-)